Formula/libgcrypt.rb

```diff
@@ -41,6 +41,12 @@ def install
       MachO::Tools.change_install_name("#{buildpath}/tests/.libs/random",
                                        "#{lib}/libgcrypt.20.dylib",
                                        "#{buildpath}/src/.libs/libgcrypt.20.dylib")
+
+      # The above operation modified tests/.libs/random and
+      # broke the validity of its code signature. Have to
+      # re-sign the binary so that it can be authorized to
+      # run on Apple Silicon.
+      system("codesign", "-s", "-", "#{buildpath}/tests/.libs/random")
```

Comment:
```suggestion
      MachO.codesign("#{buildpath}/tests/.libs/random")
```

Reply:
Can you try this? It should work, I think

Reply:
```
Error: An exception occurred within a child process:
  NoMethodError: undefined method `codesign' for MachO:Module
Did you mean?  codesign!
```

Reply:
After I change it to
```ruby
MachO.codesign!("#{buildpath}/tests/.libs/random")
```
I am getting
```
==> make
==> codesign -s - /private/tmp/libgcrypt-20201211-74077-1lpxxvf/libgcrypt-1.8.7/
Last 15 lines from /Users/brett/Library/Logs/Homebrew/libgcrypt/03.codesign:
2020-12-11 18:56:06 +0800

codesign
-s
-
/private/tmp/libgcrypt-20201211-74077-1lpxxvf/libgcrypt-1.8.7/tests/.libs/random

/private/tmp/libgcrypt-20201211-74077-1lpxxvf/libgcrypt-1.8.7/tests/.libs/random: is already signed

Do not report this issue to Homebrew/brew or Homebrew/core!
```